bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/client/ConsumedThingImpl.java

```diff
@@ -0,0 +1,268 @@
+     * @param webthingUrl the identifier of a WebThing resource
+     * @param executor executor to use
+     * @param errorHandler the error handler
+     * @param webSocketConnectionFactory the Websocket connectino fctory to be used
+     * @throws IOException if the WebThing can not be connected
+     */
+    ConsumedThingImpl(HttpClient httpClient, URI webthingUrl, ScheduledExecutorService executor,
+            Consumer<String> errorHandler, WebSocketConnectionFactory webSocketConnectionFactory) throws IOException {
+        this(httpClient, webthingUrl, executor, errorHandler, webSocketConnectionFactory, DEFAULT_PING_PERIOD);
+    }
+
+    /**
+     * constructor
+     *
+     * @param httpClient the http client to use
+     * @param webthingUrl the identifier of a WebThing resource
+     * @param executor executor to use
+     * @param errorHandler the error handler
+     * @param webSocketConnectionFactory the Websocket connectino fctory to be used
+     * @param pingPeriod the ping period tothe the healthiness of the connection
+     * @throws IOException if the WebThing can not be connected
+     */
+    ConsumedThingImpl(HttpClient httpClient, URI webthingUrl, ScheduledExecutorService executor,
+            Consumer<String> errorHandler, WebSocketConnectionFactory webSocketConnectionFactory, Duration pingPeriod)
+            throws IOException {
+        this.webThingURI = webthingUrl;
+        this.httpClient = httpClient;
+        this.errorHandler = errorHandler;
+        this.description = new DescriptionLoader(httpClient).loadWebthingDescription(webThingURI,
+                Duration.ofSeconds(20));
+
+        // opens a websocket downstream to be notified if a property value will be changed
+        this.websocketDownstream = webSocketConnectionFactory.create(this.getEventStreamUri(), executor, this::onError,
+                pingPeriod);
+    }
+
+    private Optional<URI> getPropertyUri(String propertyName) {
+        var optionalProperty = description.getProperty(propertyName);
+        if (optionalProperty.isPresent()) {
+            var propertyDescription = optionalProperty.get();
+            for (var link : propertyDescription.links) {
+                if ((link.rel != null) && (link.href != null) && link.rel.equals("property")) {
+                    return Optional.of(webThingURI.resolve(link.href));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private URI getEventStreamUri() {
+        for (var link : this.description.links) {
+            var href = link.href;
+            if (href != null) {
+                var rel = Optional.ofNullable(link.rel).orElse("<undefined>");
+                if (rel.equals("alternate")) {
+                    return URI.create(href);
+                }
+            }
+        }
+        throw new IllegalStateException("WebThing " + webThingURI
+                + " does not support websocket uri. WebThing description: " + this.description);
```

Comment:
When is this exception raised? If I see correctly it is not cought anywhere. That means your binding will crash if it is raised.

Reply:
The exception will be thrown if the webthing does not support a web socket uri (which should). I have changed the exception type to IOException which will be handled inside the WebThingHandler class. Thanks for the hint